src/main/java/esi/backend/controller/RequestController.java

```diff
@@ -23,71 +17,39 @@
 public class RequestController {
 
     private final RequestService requestService;
-    private final CarService carService;
-    private final RentalService rentalService;
-    private final CustomerRepository customerRepository;
 
-    public RequestController(RequestService requestService, CarService carService, RentalService rentalService, CustomerRepository customerRepository) {
+    public RequestController(RequestService requestService) {
         this.requestService = requestService;
-        this.carService = carService;
-        this.rentalService = rentalService;
-        this.customerRepository = customerRepository;
     }
 
     @RequestMapping("requests")
     @PreAuthorize("hasRole('MANAGER')")
-    public List<Request> getAllRequests() {
+    public ResponseEntity<List<Request>> getAllRequests() {
         return requestService.getAllRequests();
     }
 
     @RequestMapping("cars/{carId}/requests")
     @PreAuthorize("hasRole('MANAGER')")
-    public List<Request> getAllRequestsByCarId(@PathVariable UUID carId) {
+    public ResponseEntity<List<Request>> getAllRequestsByCarId(@PathVariable UUID carId) {
         return requestService.getAllRequestsByCarId(carId);
     }
 
     @RequestMapping("cars/{carId}/requests/{id}")
     @PreAuthorize("hasRole('MANAGER')")
-    public Optional<Request> getRequest(@PathVariable UUID id) {
+    public ResponseEntity<Request> getRequest(@PathVariable UUID id) {
         return requestService.getRequest(id);
     }
 
     @RequestMapping(method = RequestMethod.POST, value = "/cars/{carId}/requests")
     @PreAuthorize("hasRole('CUSTOMER')")
     public ResponseEntity<?> addRequest(@AuthenticationPrincipal final UserDetails currentUser, @RequestBody Request request, @PathVariable UUID carId) {
-        Optional<Car> optionalCar = carService.getCar(carId);
-        if (optionalCar.isPresent()) {
-            request.setCar(optionalCar.get());
-        } else throw new ResourceNotFoundException("Car with id" + carId + "not found");
-        Optional<Customer> optionalCustomer = customerRepository.findByUsername(currentUser.getUsername());
-        optionalCustomer.ifPresent(request::setCustomer);
-        requestService.addRequest(request);
-        return ResponseEntity.ok("Request created successfully!");
+        requestService.addRequest(currentUser, request, carId);
```

Comment:
`carId` is passed as the 3rd argument here, but then in the `RequestService` `updateRequest` method, it is used for retrieving the request and checking if it exists.